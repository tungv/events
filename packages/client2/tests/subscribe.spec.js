import execa from 'execa';
import path from 'path';

const subscribe = async ({ loglevel = 'DEBUG', configPath, json = true }) => {
  const child = execa('node', ['-r', 'babel-register', './src/subscribe'], {
    env: {
      params: JSON.stringify({
        json,
        verbose: loglevel,
        configPath: path.resolve(process.cwd(), configPath),
      }),
    },
  });

  const { stdout } = await child;
  return { stdout };
};

const normalize = array =>
  array.map(item => {
    const obj = JSON.parse(item.split(process.cwd()).join('<ROOT>'));
    delete obj._t;

    return obj;
  });

describe('heq-client subscribe', () => {
  it('should log', async () => {
    const { stdout } = await subscribe({
      configPath: './fixtures/config/sample.config.js',
    });

    expect(normalize(stdout.split('\n'))).toMatchSnapshot();
  });
});
