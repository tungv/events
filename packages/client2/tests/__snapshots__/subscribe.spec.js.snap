// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heq-client subscribe should log 1`] = `
Array [
  Object {
    "_l": "INFO",
    "payload": Object {
      "path": "<ROOT>/fixtures/config/test.config.js",
    },
    "type": "load-config-begin",
  },
  Object {
    "_l": "DEBUG",
    "payload": Object {
      "config": Object {
        "hotReload": Object {
          "watchPaths": Array [
            "<ROOT>",
          ],
        },
        "monitor": Object {
          "port": 43333,
        },
        "persist": Object {
          "driver": "@events/snapshot-mongo",
          "seedFilePath": false,
          "store": "mongodb://localhost/client_test",
        },
        "sideEffects": Object {
          "sideEffectsPath": "<ROOT>/fixtures/sideEffects/sampleEffects.js",
        },
        "subscribe": Object {
          "burstCount": 20,
          "burstTime": 500,
          "maxRetry": 20,
          "serverUrl": "http://localhost:43366",
        },
        "transform": Object {
          "rulePath": "<ROOT>/fixtures/rules/user_management.js",
        },
      },
    },
    "type": "load-config-end",
  },
  Object {
    "_l": "DEBUG",
    "payload": Object {
      "rulePath": "<ROOT>/fixtures/rules/user_management.js",
    },
    "type": "load-transform-begin",
  },
  Object {
    "_l": "DEBUG",
    "payload": Object {
      "ruleMeta": Array [
        Object {
          "name": "users",
          "version": "1.0.0",
        },
        Object {
          "name": "departments",
          "version": "1.0.0",
        },
      ],
    },
    "type": "load-transform-end",
  },
  Object {
    "_l": "DEBUG",
    "payload": Object {
      "persistenceConfig": Object {
        "driver": "@events/snapshot-mongo",
        "seedFilePath": false,
        "store": "mongodb://localhost/client_test",
      },
    },
    "type": "connect-snapshot-begin",
  },
  Object {
    "_l": "INFO",
    "payload": Object {
      "snapshotVersion": 0,
    },
    "type": "connect-snapshot-end",
  },
  Object {
    "_l": "DEBUG",
    "payload": Object {
      "retryCount": 0,
    },
    "type": "connect-events-begin",
  },
  Object {
    "_l": "INFO",
    "payload": Object {},
    "type": "connect-events-end",
  },
  Object {
    "_l": "INFO",
    "type": "process-interrupted",
  },
  Object {
    "_l": "INFO",
    "payload": Object {
      "code": 0,
    },
    "type": "process-exit",
  },
]
`;
